server/pulp/server/controllers/consumer.py

```diff
@@ -46,31 +40,29 @@ def unbind(consumer_id, repo_id, distributor_id, options):
     :type distributor_id: str
     :param options: Unbind options passed to the handler.
     :type options: dict
-    :returns TaskResult containing the result of the unbind & any spawned tasks or a dictionary
-             of the unbind result if no tasks were spawned.
-    :rtype: TaskResult
+    :returns Dictionary containing the result of the unbind.
+    :rtype: dict
     """
 
     bind_manager = managers.consumer_bind_manager()
     binding = bind_manager.get_bind(consumer_id, repo_id, distributor_id)
-
-    response = TaskResult(result=binding)
-
     bind_manager.delete(consumer_id, repo_id, distributor_id, True)
-
-    return response
+    return binding
 
 
 def force_unbind(consumer_id, repo_id, distributor_id, options):
```

Comment:
Maybe out of scope for this PR, but I don't think we need this behavior anymore. Normally a consumer would be notified of an unbinding, and the agent would perform necessary changes on the consumer. This function was intended for a case where a consumer was not available to participate in its own unbinding.